libres/lib/include/ert/analysis/update.hpp

```diff
@@ -9,6 +9,43 @@
 #include <ert/enkf/enkf_obs.hpp>
 
 namespace analysis {
+class update_data_type : public std::enable_shared_from_this<update_data_type> {
+    /**
```

Comment:
Docstring should be before the class. As in, between `namespace analysis` and `class update_data_type`.